src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -256,6 +267,21 @@ protected Object readResolve() {
         return o;
     }
 
+    protected void setInstanceKeypair(KeyPair keypair) {
```

Comment:
This does not appear to ever be called?

Reply:
yes, i missed a [commit](https://github.com/jenkinsci/ec2-plugin/pull/993/commits/4a3e23c7b5e271d0df45fe4a809631dbaeb2731d) :facepalm: